packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -509,28 +550,47 @@ T _runSync<T>(T Function() op, {
 ///
 /// See also:
 ///   * [ErrorHandlingFileSystem], for a similar file system strategy.
-class ErrorHandlingProcessManager extends ProcessManager {
+class ErrorHandlingProcessManager implements ProcessManager {
   ErrorHandlingProcessManager({
-    @required ProcessManager delegate,
     @required Platform platform,
-  }) : _delegate = delegate,
-       _platform = platform;
-
-  final ProcessManager _delegate;
+    @required FileSystem fileSystem,
+    @required Logger logger,
+    @visibleForTesting ProcessKillPid killPid = io.Process.killPid,
+    @visibleForTesting ProcessRun run = io.Process.run,
+    @visibleForTesting ProcessRunSync runSync = io.Process.runSync,
+    @visibleForTesting ProcessStart start = io.Process.start,
+  }) : _platform = platform,
+       _fileSytem = fileSystem,
+       _logger = logger,
+       _killPid = killPid,
+       _processRun = run,
+       _processRunSync = runSync,
+       _processStart = start;
+
+  final Logger _logger;
+  final FileSystem _fileSytem;
   final Platform _platform;
+  final ProcessKillPid _killPid;
+  final ProcessRun _processRun;
+  final ProcessRunSync _processRunSync;
+  final ProcessStart _processStart;
+
+  // Cached executable paths, the first is the working directory, the next is a map from
+  // executable name to actual path.
+  final Map<String, Map<String, String>> _resolvedExecutables = <String, Map<String, String>>{};
 
   @override
   bool canRun(dynamic executable, {String workingDirectory}) {
-    return _runSync(
-      () => _delegate.canRun(executable, workingDirectory: workingDirectory),
+    return _runSync<bool>(
+      () => _getCommandPath(<dynamic>[executable], workingDirectory, strict: true) != null,
```

Comment:
Needed to re-add the `strict` flag since by default we return the executable path if it cannot be found